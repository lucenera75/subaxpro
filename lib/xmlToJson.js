import xml2js from 'xml2js';

export default async function (xmlData) {
	try {
		const parsedXmlData = await xml2js.parseStringPromise(xmlData);
		return {
			id: parsedXmlData.Data.id[0],
			name: parsedXmlData.Data.name[0],
			description: parsedXmlData.Data.description[0],
		};
	} catch (err) {
		throw new Error('Unable to parse correctly xmlData');
	}
}
