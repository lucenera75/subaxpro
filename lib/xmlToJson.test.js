import xmlToJson from './xmlToJson';

test('correct company xmlData matches snapshot', async () => {
	const result = await xmlToJson(`
	<?xml version="1.0" encoding="UTF-8"?>
	<Data>
		<id>1</id>
		<name>MWNZ</name>
		<description>..is awesome</description>
	</Data>
	`);
	expect(result).toMatchSnapshot();
});

test('non correct company xmlData throws an error and does not return a result', async () => {
	let result;
	try {
		result = await xmlToJson(`
		SOME RANDOM STUFF
		`);
	} catch (err) {
		expect(err.message).toMatchSnapshot();
	}
	expect(result).toBeUndefined();
});
