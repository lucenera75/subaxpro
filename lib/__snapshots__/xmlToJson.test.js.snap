// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct company xmlData matches snapshot 1`] = `
Object {
  "description": "..is awesome",
  "id": "1",
  "name": "MWNZ",
}
`;

exports[`non correct company xmlData throws an error and does not return a result 1`] = `"Unable to parse correctly xmlData"`;
